exchanges/binance/binance_wrapper.go

```diff
@@ -821,55 +1338,181 @@ func (b *Binance) GetOrderHistory(req *order.GetOrdersRequest) ([]order.Detail,
 	if err := req.Validate(); err != nil {
 		return nil, err
 	}
-
 	if len(req.Pairs) == 0 {
 		return nil, errors.New("at least one currency is required to fetch order history")
 	}
-
 	var orders []order.Detail
-	for x := range req.Pairs {
-		fpair, err := b.FormatExchangeCurrency(req.Pairs[x], asset.Spot)
-		if err != nil {
-			return nil, err
+	switch req.AssetType {
+	case asset.Spot:
+		for x := range req.Pairs {
+			fpair, err := b.FormatExchangeCurrency(req.Pairs[x], asset.Spot)
+			if err != nil {
+				return nil, err
+			}
+			resp, err := b.AllOrders(fpair.String(),
+				"",
+				"1000")
+			if err != nil {
+				return nil, err
+			}
+			for i := range resp {
+				orderSide := order.Side(strings.ToUpper(resp[i].Side))
+				orderType := order.Type(strings.ToUpper(resp[i].Type))
+				orderDate, err := convert.TimeFromUnixTimestampFloat(resp[i].Time)
+				if err != nil {
+					return nil, err
+				}
+				// New orders are covered in GetOpenOrders
+				if resp[i].Status == "NEW" {
+					continue
+				}
+				pair, err := currency.NewPairFromString(resp[i].Symbol)
+				if err != nil {
+					return nil, err
+				}
+				orders = append(orders, order.Detail{
+					Amount:   resp[i].OrigQty,
+					Date:     orderDate,
+					Exchange: b.Name,
+					ID:       strconv.FormatInt(resp[i].OrderID, 10),
+					Side:     orderSide,
+					Type:     orderType,
+					Price:    resp[i].Price,
+					Pair:     pair,
+					Status:   order.Status(resp[i].Status),
+				})
+			}
 		}
-		resp, err := b.AllOrders(fpair.String(),
-			"",
-			"1000")
-		if err != nil {
-			return nil, err
+	case asset.CoinMarginedFutures:
+		var orderHistory []FuturesOrderData
+		var err error
+		var pairs currency.Pairs
+		if len(req.Pairs) > 0 {
+			pairs = req.Pairs
+		} else {
+			pairs, err = b.GetEnabledPairs(asset.CoinMarginedFutures)
+			if err != nil {
+				return orders, err
+			}
 		}
-
-		for i := range resp {
-			orderSide := order.Side(strings.ToUpper(resp[i].Side))
-			orderType := order.Type(strings.ToUpper(resp[i].Type))
-			orderDate, err := convert.TimeFromUnixTimestampFloat(resp[i].Time)
+		for i := range pairs {
+			fPair, err := b.FormatExchangeCurrency(pairs[i], req.AssetType)
 			if err != nil {
-				return nil, err
+				return orders, err
 			}
-			// New orders are covered in GetOpenOrders
-			if resp[i].Status == "NEW" {
-				continue
+			var complete bool
+			for !complete {
+				orderHistory, err = b.GetAllFuturesOrders(fPair.String(), "", req.StartTicks, req.EndTicks, 0, 100)
```

Comment:
Since I'm a super hardcore trader, I get 0 orders back here. However, you'll need to do length checks below, otherwise you'll get panics